crates/ruma-events-macros/src/event_parse.rs

```diff
@@ -226,13 +237,36 @@ impl Parse for EventEnumInput {
     }
 }
 
-pub struct EventEnumEntry {
-    pub attrs: Vec<Attribute>,
-    pub ev_type: LitStr,
+/// The entire `event_enum!` macro structure directly as it appears in the source code.
+pub struct EventEnumInputs {
+    pub(crate) enums: Vec<EventEnumInput>,
 }
 
-impl Parse for EventEnumEntry {
-    fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
-        Ok(Self { attrs: input.call(Attribute::parse_outer)?, ev_type: input.parse()? })
+impl Parse for EventEnumInputs {
+    fn parse(input: ParseStream<'_>) -> parse::Result<Self> {
+        let mut enums = vec![];
+        loop {
+            let attrs = input.call(Attribute::parse_outer)?;
+
+            if input.parse::<Token![enum]>().is_err() {
+                break;
+            };
+
+            let name = input.parse::<EventKind>()?;
```

Comment:
I would prefer consistently not using turbofishes for `parse`:

```suggestion
            let name: EventKind = input.parse()?;
```